js/ui/map.js

```diff
@@ -197,23 +197,11 @@ var Map = module.exports = function(options) {
         this.style.update(this._classes, {transition: false});
     });
 
-    this.on('style.change', function() {
+    this.on('data', function() {
         this._update(true);
```

Comment:
Good catch. Fortunately it looks like filtering on `event.dataType === 'style'` is equivalent to listening to the `style.change` event.